lib/kennel/syncer.rb

```diff
@@ -256,11 +256,15 @@ def prevent_irreversible_partial_updates
     def populate_id_map(expected, actual)
       expected.each do |e|
         @id_map.set_new(e.class.api_resource, e.tracking_id)
+        if e.class.api_resource == "synthetics/tests"
+          @id_map.set_new(Kennel::Models::Monitor.api_resource, e.tracking_id)
+        end
       end
 
       actual.each do |a|
         next unless tracking_id = a.fetch(:tracking_id)
-        next unless @id_map.get(a.fetch(:klass).api_resource, tracking_id)
+        next unless @id_map.get(a.fetch(:klass).api_resource, tracking_id) \
+          || (@project_filter && !tracking_id.start_with?("#{@project_filter}:"))
```

Comment:
should we just remove this and always set it ?

Reply:
the actual is the source of truth after all 🤷

Reply:
```
  1) Failure:
Kennel::Syncer::#update#test_0009_fails on to-be-deleted dependencies [/Users/revans/git/github.com/grosser/kennel/test/kennel/syncer_test.rb:557]:
unexpected invocation: #<Mock:Api>.create("slo", {:name => "x🔒", :description => "x\n-- Managed by kennel a:c in test/test_helper.rb, do not modify manually", :thresholds => [], :monitor_ids => [456], :tags => ["service:a", "team:test_team"], :type => "metric"})
```

If we were to remove this line then we'd either need to find another fix, or consciously not care about this behaviour.

Reply:
"Only use this actual (i.e. make it available for resolution) if it either corresponds to an expected, or ( we're using a project filter and it's from a different project )"

Reply:
> the actual is the source of truth after all 🤷

Not when you're using a project filter it isn't.

Reply:
ok, get it now ... not perfect but good enough ...